File: sdks/go/pkg/models/shared/groupmembership.go
package shared

import (
	"time"
)

type GroupMembershipRoleEnum string

const (
	GroupMembershipRoleEnumMember GroupMembershipRoleEnum = "member"
	GroupMembershipRoleEnumAdmin  GroupMembershipRoleEnum = "admin"
)

type GroupMembershipTypeEnum string

const (
	GroupMembershipTypeEnumGroupMembership GroupMembershipTypeEnum = "group_membership"
)

// GroupMembership
// Membership is used to signify that a user is part of a
// group.
type GroupMembership struct {
	CreatedAt  *time.Time               `json:"created_at,omitempty"`
	Group      *GroupMini               `json:"group,omitempty"`
	ID         *string                  `json:"id,omitempty"`
	ModifiedAt *time.Time               `json:"modified_at,omitempty"`
	Role       *GroupMembershipRoleEnum `json:"role,omitempty"`
	Type       *GroupMembershipTypeEnum `json:"type,omitempty"`
	User       *UserMini                `json:"user,omitempty"`
}

File: sdks/java/lib/src/main/java/models/shared/GroupMembership.java
package .models.shared;

import java.time.OffsetDateTime;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonInclude.Include;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import .utils.DateTimeSerializer;
import .utils.DateTimeDeserializer;

/**
 * GroupMembership
 * Membership is used to signify that a user is part of a
 * group.
**/
public class GroupMembership {
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("created_at")
    public OffsetDateTime createdAt;
    public GroupMembership withCreatedAt(OffsetDateTime createdAt) {
        this.createdAt = createdAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("group")
    public GroupMini group;
    public GroupMembership withGroup(GroupMini group) {
        this.group = group;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("id")
    public String id;
    public GroupMembership withId(String id) {
        this.id = id;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonSerialize(contentUsing = DateTimeSerializer.class)
    @JsonDeserialize(contentUsing = DateTimeDeserializer.class)
    @JsonProperty("modified_at")
    public OffsetDateTime modifiedAt;
    public GroupMembership withModifiedAt(OffsetDateTime modifiedAt) {
        this.modifiedAt = modifiedAt;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("role")
    public GroupMembershipRoleEnum role;
    public GroupMembership withRole(GroupMembershipRoleEnum role) {
        this.role = role;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("type")
    public GroupMembershipTypeEnum type;
    public GroupMembership withType(GroupMembershipTypeEnum type) {
        this.type = type;
        return this;
    }
    @JsonInclude(Include.NON_ABSENT)
    @JsonProperty("user")
    public UserMini user;
    public GroupMembership withUser(UserMini user) {
        this.user = user;
        return this;
    }
}